### PR TITLE
Fix integer overflow bug in triu/tril for large diagonal values

### PR DESCRIPTION
**Pull request by PenXLa**
This PR fixes a bug in the implementation of `apply_triu_tril_single` where using extremely large values for the diagonal argument (e.g. `diagonal=9223372036854775807`) could result in integer overflow and incorrect results. The masking logic is re-written to avoid this issue by always iterating over all columns, ensuring correctness even for large or extreme diagonal values.

Example of the original incorrect behavior:
```python
a = torch.ones(5,5)
torch.triu(a, 9223372036854775807)
# Before:
# tensor([[0., 0., 0., 0., 0.],
#         [1., 1., 1., 1., 1.],
#         [1., 1., 1., 1., 1.],
#         [1., 1., 1., 1., 1.],
#         [1., 1., 1., 1., 1.]])
```

The new implementation guards against overflow and produces correct results for all valid input values.

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Would it be simpler for this to branch earlier actually? If the diagonal argument is larger than the dimension, we can return a matrix full of zeros or a clone of the input directly?

**Comment by PenXLa** [comment]
> Would it be simpler for this to branch earlier actually? If the diagonal argument is larger than the dimension, we can return a matrix full of zeros or a clone of the input directly?

Thanks for the suggestion — that does make sense.

One alternative I considered is clamping k before the computation, like so:

```cpp
k = std::clamp(k, -n, m); // Clamp k to [-n, m] to prevent i + k arithmetic overflow, especially if k approaches INT64_MAX/INT64_MIN.
```

This would avoid the overflow risk without requiring additional early branching logic, and still lead to returning an empty or cloned matrix where appropriate. Let me know what you think — happy to go either way.

**Review by albanD — COMMENTED** [review]
Thanks for the fix!
Would you be able to add a test case to make sure this behaves as expected?

**Comment by PenXLa** [comment]
> Thanks for the fix! Would you be able to add a test case to make sure this behaves as expected?

Sure! Just added a test case for this.

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by PenXLa** [comment]
Hi @albanD, the lint issue is fixed.
